extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -194,6 +197,52 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
         return getterAccessorsContainer;
     }
 
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateClassHasFieldId(IndexView index, String entityType) {
+        // create a new independent class name that we can override
+        DotName className = DotName.createSimple(entityType);
+        ClassInfo classInfo = index.getClassByName(className);
+
+        if (classInfo != null) {
+            validateRec(index, entityType, classInfo);
+        }
+    }
+
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateRec(IndexView index, String entityType, ClassInfo classInfo) {
+        List<FieldInfo> fieldsNamedId = classInfo.fields().stream()
+                .filter(f -> f.name().equals("id"))
+                .collect(Collectors.toList());
+        List<AnnotationInstance> fieldsAnnotatedWithId = classInfo.annotations().stream()
+                .filter(a -> a.name().toString().endsWith("persistence.Id"))
```

Comment:
This is looking for annotations found in the class, which includes field annotations, but also method annotations, if you are looking for field annotations you should filter them. Note that you can also find them by traversing the fields, otherwise.